### ISSUE TITLE
private types and "a variant case which is never constructed" warning

### ISSUE THREAD

**Issue by johnridesabike** [reason-association/reanalyze#113]
If you have a private variant type, then it will always trigger a "variant case which is never constructed" warning. Of course, because it's private, it *can't* be constructed outside of the implementation.

**Comment by cristianoc** [comment]
Would you explain the use case a bit more. Is it constructed in the implementation, and is the function that constructs it public?

**Comment by johnridesabike** [comment]
Here’s a quick example:

```res
// Types.res
type t = A | B
let a = () => A
let b = () => B
```

```res
// Types.resi
type t = private A | B
let a: unit => t
let b: unit => t
```

```res
// Consumer.res
let f = (x: Types.t) =>
  switch x {
  | A => Js.log("a")
  | B => Js.log("b")
  }

Types.a()->f
Types.b()->f
```

```
$ yarn reanalyze -dce                                                                                                                                                                    
yarn run v1.22.10
$ ~/code/private-types/node_modules/.bin/reanalyze -dce

  Warning Dead Type
  File “~/code/private-types/Types.resi", line 1, characters 17-18
  t.A is a variant case which is never constructed
  <-- line 1
  type t = private | @dead("t.A") A | B

  Warning Dead Type
  File “~/code/private-types/Types.resi", line 1, characters 19-22
  t.B is a variant case which is never constructed
  <-- line 1
  type t = private | @dead("t.A") A | @dead("t.B") B
  
  Analysis reported 2 issues (Warning Dead Type:2)
```

**Comment by cristianoc** [comment]
Can you check with the latest version of reanalyze? Can't repro the results on the example: no warning reported.

**Comment by johnridesabike** [comment]
That’s odd. I’m on 2.13.0. Here’s a sample repo: https://github.com/johnridesabike/private-types It has the same ReScript code as above, but maybe there’s something else wrong with the setup?

**Comment by cristianoc** [comment]
@johnridesabike thanks for the repro.
Normally, the unused variant cases in the interface are a subset of those in the implementation. The issue was that some dependencies were lost because of a bug when the last file processed (`Types.resi`) is an interface file.
It should have reported zero issues on the example.

**Comment by cristianoc** [comment]
Fixed in the upcoming version 2.14.0.
